### PR TITLE
FQA update

### PR DESCRIPTION
**Pull request by AbhinandanParashar**
FAQ update

### CODE REVIEW

**Review by deepanshuHarness — APPROVED** [review]

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 1]
```diff
@@ -7011,3 +7011,23 @@ Go to [Git Repository Merge PRs](https://developer.harness.io/docs/code-reposito
 Yes, users can now store the terraform plan on the delegate and leverage it in the apply step. This now bypasses the restriction to store the plan in a secrets manager and let users store it locally.
 This feature is behind the feature flag, `CDS_STORE_TERRAFORM_PLAN_FILE_LOCALLY_ON_DELEGATE`. Harness Delegate version 827xx or later is required for this feature.
 Go to [Store Terraform Plan on Harness Delegate](https://developer.harness.io/docs/continuous-delivery/cd-infrastructure/terraform-infra/run-a-terraform-plan-with-the-terraform-plan-step/#store-terraform-plan-on-harness-delegate) and [Demo Video](https://www.loom.com/share/bc5a4f382d584b228b4ea2c82eb94a7c?sid=b9fac5c3-c11b-4f50-acff-f4fd2b3cc83a) for more information.
+
+####  We have added tags in pipeline but for some reason, in the execution the tags are not present for some reason.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7015]:
```suggestion
####  We have added tags in the pipeline but, during execution, the tags are not present.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 2]
```diff
@@ -7011,3 +7011,23 @@ Go to [Git Repository Merge PRs](https://developer.harness.io/docs/code-reposito
...
+
+The issue can be with the terraform, where in the case of pipeline creation via tf, we need to define tags in the template and the in the tf resource to get it applied properly.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7017]:
```suggestion
The issue can be with Terraform. When you create a pipeline with Terraform, you must define tags in the template as well as the Terraform resource to get it applied properly.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 3]
```diff
@@ -7011,3 +7011,23 @@ Go to [Git Repository Merge PRs](https://developer.harness.io/docs/code-reposito
...
+
+#### When running a terraform, this pipeline is pulling a previous state from another pipeline, what could be the issue.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7019]:
```suggestion
#### When running Terraform, pipeline is pulling a previous state from another pipeline, what could be the issue?
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 4]
```diff
@@ -7011,3 +7011,23 @@ Go to [Git Repository Merge PRs](https://developer.harness.io/docs/code-reposito
...
+
+If you are running the pipeline on the same delegate make sure Provisioner Identifier is different for both the plan.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7021]:
```suggestion
If you are running the pipeline on the same delegate, make sure that Provisioner Identifier is different for both plans.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 5]
```diff
@@ -7011,3 +7011,23 @@ Go to [Git Repository Merge PRs](https://developer.harness.io/docs/code-reposito
...
+
+#### We have a updated manifest file for deployment, but delegate seems to be fetching old manifest how can we update this.
+
+You can clear the local cached repo : Local repository is stored at :
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7025]:
```suggestion
You can clear the local cached repo. 
Local repository is stored at : 
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 6]
```diff
@@ -7011,3 +7011,23 @@ Go to [Git Repository Merge PRs](https://developer.harness.io/docs/code-reposito
...
+If you are running the pipeline on the same delegate make sure Provisioner Identifier is different for both the plan.
+
+#### We have a updated manifest file for deployment, but delegate seems to be fetching old manifest how can we update this.
+
+You can clear the local cached repo : Local repository is stored at : 
+
+```
+/opt/harness-delegate/repository/gitFileDownloads/Nar6SP83SJudAjNQAuPJWg/<connector-id>/<repo-name>/<sha1-hash-of-repo-url>
+```
+#### We are facing missing terraform backend config issue in terraform plan step logs, though we have configured backend.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7030]:
```suggestion
#### We are facing missing Terraform backend configuration issue in the Terraform Plan step logs though we have configured backend.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 7]
```diff
@@ -7011,3 +7011,23 @@ Go to [Git Repository Merge PRs](https://developer.harness.io/docs/code-reposito
...
+
+You have to declare the backend block in terraform configuration. You might have provided in harness backend-configuration, but in your terraform configuration doesn’t exist any terraform “backend” block.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:7032]:
```suggestion
You have to declare the backend block in the Terraform configuration. You might have provided the config in Harness, but the `backend` block might not exist in the Terraform configuration.
```

File: kb/platform/harness-platform-faqs.md [hunk 8]
```diff
@@ -2959,3 +2959,14 @@ By following these steps, our support team can promptly review the situation, di
 
 "Exit code 137" typically indicates an out-of-memory error. When a process in a system exhausts its allocated memory resources, the operating system sends a termination signal to the process. In the case of "Exit code 137," this signal signifies that the process was terminated due to running out of memory. This error commonly occurs when a program or container attempts to allocate more memory than is available, leading to termination by the system to prevent resource exhaustion and potential system instability.
 
+### How to install latest version of azure cli in our Harness Dlegates.
```
Comment (roshnisarangadharan) [inline kb/platform/harness-platform-faqs.md:2962]:
```suggestion
### How to install the latest version of Azure CLI in Harness Delegate?
```

File: kb/platform/harness-platform-faqs.md [hunk 9]
```diff
@@ -2959,3 +2959,14 @@ By following these steps, our support team can promptly review the situation, di
...
+
+You can follow:
+
+```
+## Install Azure CLI
+rpm --import <https://packages.microsoft.com/keys/microsoft.asc>
+rpm -ivh <https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm>
+microdnf install -y azure-cli
+
+```
```
Comment (roshnisarangadharan) [inline kb/platform/harness-platform-faqs.md:2972]: Shouldn't you mention locate the `INIT_SCRIPT` in the `delegate.yaml` and replace value"" with the following microdnf script? Something like this, https://developer.harness.io/docs/continuous-delivery/deploy-srv-diff-platforms/tanzu/tanzu-app-services-quickstart#install-cloud-foundry-command-line-interface-cf-cli-on-your-harness-delegate.

File: kb/security-testing-orchestration/security-testing-orchestration-faqs.md [hunk 10]
```diff
@@ -87,3 +87,7 @@ Go to [Troubleshoot Sonar Scans](/docs/security-testing-orchestration/sto-techre
 
 #### Why am I getting the error Missing target_name for scan_type [repository] scan.
 This error ocurrs if there's no scan target in the Scanner configuration. To fix this, please ensure that the Scan Step configuration properly selects a target.
+
+#### I'm trying to create a dashboard to display the STO metrics of my project. But my Organization is not getting listed in the filter, hence I'm not able to get my project filtered as well.
```
Comment (roshnisarangadharan) [inline kb/security-testing-orchestration/security-testing-orchestration-faqs.md:91]:
```suggestion
#### I'm trying to create a dashboard to display the STO metrics for my project. But, my Organization is not getting listed in the filter. Hence, I'm not able to get my project filtered as well.
```

File: kb/security-testing-orchestration/security-testing-orchestration-faqs.md [hunk 11]
```diff
@@ -87,3 +87,7 @@ Go to [Troubleshoot Sonar Scans](/docs/security-testing-orchestration/sto-techre
...
+
+The dashboard only populates projects that have had a baseline scan run. You will need to add baseline to pipelines which needs to be displayed in dashboard.
```
Comment (roshnisarangadharan) [inline kb/security-testing-orchestration/security-testing-orchestration-faqs.md:93]:
```suggestion
The dashboard only populates projects that had a baseline scan run. You might want to add baseline to pipelines to display them in the dashboard.
```

File: kb/platform/harness-platform-faqs.md [hunk 12]
```diff
@@ -2959,3 +2959,14 @@ By following these steps, our support team can promptly review the situation, di
 
 "Exit code 137" typically indicates an out-of-memory error. When a process in a system exhausts its allocated memory resources, the operating system sends a termination signal to the process. In the case of "Exit code 137," this signal signifies that the process was terminated due to running out of memory. This error commonly occurs when a program or container attempts to allocate more memory than is available, leading to termination by the system to prevent resource exhaustion and potential system instability.
 
+### How to install latest version of azure cli in our Harness Dlegates.
+
+You can follow:
```
Comment (roshnisarangadharan) [inline kb/platform/harness-platform-faqs.md:2964]:
```suggestion
* Locate the `INIT_SCRIPT` in the `Deployment` object of the `delegate.yaml` file.
   ```
   - name: INIT_SCRIPT  
   value: ""  
   ```
* Replace `value: "" ` with the following:
```

File: kb/platform/harness-platform-faqs.md (hunk 9 above)
Comment (roshnisarangadharan) [inline kb/platform/harness-platform-faqs.md:2972]:
```suggestion
* Locate the `INIT_SCRIPT` in the `Deployment` object of the `delegate.yaml` file.
   ```
   - name: INIT_SCRIPT  
   value: ""  
   ```
* Replace `value: "" ` with the following:

  ```
  ## Install Azure CLI
  rpm --import <https://packages.microsoft.com/keys/microsoft.asc>
  rpm -ivh <https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm>
  microdnf install -y azure-cli
  ```
```

File: kb/security-testing-orchestration/security-testing-orchestration-faqs.md [hunk 13]
```diff
@@ -87,3 +87,7 @@ Go to [Troubleshoot Sonar Scans](/docs/security-testing-orchestration/sto-techre
 
 #### Why am I getting the error Missing target_name for scan_type [repository] scan.
 This error ocurrs if there's no scan target in the Scanner configuration. To fix this, please ensure that the Scan Step configuration properly selects a target.
+
+#### I'm trying to create a dashboard to display the STO metrics for my project. But, my Organization is not getting listed in the filter. Hence, I'm not able to get my project filtered as well.
```
Comment (roshnisarangadharan) [inline kb/security-testing-orchestration/security-testing-orchestration-faqs.md:91]:
```suggestion
#### I'm trying to create a dashboard to display the STO metrics for my project. But, my organization is not getting listed in the filter. Hence, I'm not able to get my project filtered as well.
```

File: kb/security-testing-orchestration/security-testing-orchestration-faqs.md [hunk 14]
```diff
@@ -87,3 +87,7 @@ Go to [Troubleshoot Sonar Scans](/docs/security-testing-orchestration/sto-techre
...
+
+The dashboard only populates projects that had a baseline scan run. You might want to add baseline to pipelines to display them in the dashboard.
```
Comment (roshnisarangadharan) [inline kb/security-testing-orchestration/security-testing-orchestration-faqs.md:93]:
```suggestion
The dashboard only populates projects that had a baseline scan run before. You might want to add baseline to pipelines to display them in the dashboard.
```

**Review by roshnisarangadharan — DISMISSED** [review]

File: kb/platform/harness-platform-faqs.md [hunk 15]
```diff
@@ -2975,3 +2975,17 @@ By following these steps, our support team can promptly review the situation, di
 
 "Exit code 137" typically indicates an out-of-memory error. When a process in a system exhausts its allocated memory resources, the operating system sends a termination signal to the process. In the case of "Exit code 137," this signal signifies that the process was terminated due to running out of memory. This error commonly occurs when a program or container attempts to allocate more memory than is available, leading to termination by the system to prevent resource exhaustion and potential system instability.
 
+### How to install the latest version of Azure CLI in Harness Delegate?
+
+1. Locate the `INIT_SCRIPT` in the `Deployment` object of the `delegate.yaml` file.
+   ```
+   - name: INIT_SCRIPT  
+    value: ""
+  ```
```
Comment (roshnisarangadharan) [inline kb/platform/harness-platform-faqs.md:2984]:
```suggestion
   ```
```

**Review by roshnisarangadharan — APPROVED** [review]

File: kb/platform/harness-platform-faqs.md [hunk 16]
```diff
@@ -2975,3 +2975,14 @@ By following these steps, our support team can promptly review the situation, di
...
+
+Add the below commands in the `INIT_SCRIPT` to download and install the Azure CLI for the delegate.
+
+```yaml
+- name: INIT_SCRIPT
+  value: |
+    rpm --import <https://packages.microsoft.com/keys/microsoft.asc>
+    rpm -ivh <https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm>
+    microdnf install -y azure-cli
+```
```
Comment (brian-f-harness) [inline kb/platform/harness-platform-faqs.md:2988]:
```suggestion
```

**Review by brian-f-harness — COMMENTED** [review]
PL question is already covered in Delegates section line 1854. Removing from this PR.

**Review by brian-f-harness — DISMISSED** [review]
Approved for PL. Removed duplicate question.

File: kb/platform/harness-platform-faqs.md (hunk 9 above)
Comment (brian-f-harness) [reply kb/platform/harness-platform-faqs.md]: @roshnisarangadharan I removed this a a duplicate question. It is already covered in the Delegate section.

**Review by douglas-j-bothwell — APPROVED** [review]

**Pull request merged**